packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -65,15 +65,20 @@ function percentToValue(percent, min, max) {
   return (max - min) * percent + min;
 }
 
-function ensurePrecision(value, step) {
-  const stepDecimalPart = step.toString().split('.')[1];
-  const stepPrecision = stepDecimalPart ? stepDecimalPart.length : 0;
+function getDecimalPrecision(num) {
+  if (Math.abs(num) < 1) {
```

Comment:
Will we have the same problem with step=1.00000001? But maybe we can ignore it?

Reply:
since 1.00000001 is bigger than 1. It will use the old logic which works fine for big numbers.

Reply:
Oh yeah, of course!

Reply:
🤦‍♂️

Reply:
A small comment would be 👌

Reply:
Ok, I'll add a comment to explain the behavior